File: 2024/Leetcode/problem53.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        currSum = nums[0]
        maxSum = currSum
        for i in range(1, len(nums)):
            if currSum < 0:
                currSum = 0

            currSum += nums[i]
            if currSum > maxSum:
                maxSum = currSum

        return maxSum

File: 2024/Leetcode/problem53.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSum = nums[0];
        int currentSum = nums[0];
        
        for (int i = 1; i < nums.size(); i++) {
            if (currentSum < 0) {
                currentSum = 0;
            }
            currentSum += nums[i];

            if (currentSum > maxSum) {
                maxSum = currentSum;
            }
        }
        return maxSum;
    }
};